src/client/mod.rs

```diff
@@ -536,3 +559,100 @@ impl WeakClient {
         self.inner.upgrade().map(|inner| Client { inner })
     }
 }
+
+#[cfg(feature = "csfle")]
+pub struct EncryptedClientBuilder<const KMS_SET: bool> {
+    options: ClientOptions,
+    enc_options: AutoEncryptionOptions,
+}
+
+#[cfg(feature = "csfle")]
+impl<const KMS_SET: bool> EncryptedClientBuilder<KMS_SET> {
+    /// Add configuration for a single KMS provider.
+    pub fn kms_provider(
+        mut self,
+        provider: KmsProvider,
+        config: bson::Document,
+    ) -> EncryptedClientBuilder<true> {
+        self.enc_options.kms_providers.insert(provider, config);
+        EncryptedClientBuilder {
+            options: self.options,
+            enc_options: self.enc_options,
+        }
+    }
+
+    /// Add configuration for multiple KMS providers.  Already-added configuration for other
+    /// providers will be preserved.
```

Comment:
I think it would be good to have a doc-example that demonstrates this vs an equalent invocation using `kms_provider`.

Reply:
Done.